llama-index-integrations/agent/llama-index-agent-openai/llama_index/agent/openai/step.py

```diff
@@ -716,8 +716,14 @@ async def _arun_step(
             else []
         )
 
+        agent_response = AgentChatResponse(
+            response=response,
+            sources=latest_tool_outputs,
+            is_dummy_stream=(mode == ChatResponseMode.STREAM),
```

Comment:
I think this should only be set when its a return direct tool, not just when its streaming 👀

Reply:
Yes good catch! Turns out we don't need to modify much of the `agent_chat_response` stuff like I did, so I've reverted it and bypassed this issue.

Reply:
Yay!